subarray_sum_equals_k.java

```diff
@@ -0,0 +1,52 @@
+// Time Complexity : O(n), where n is the length of the input array (Iterate over the input array once to find the runningSum)
+// Space Complexity :O(n), were n is the length of input array (space required to store runningSum occurences is hashMap)
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : To find the difference between the approach used for contiguous array problem 
```

Comment:
In both problems we check whether some form of running has been seen earlier or not